### PR TITLE
Compatibility with MT 5.0.0+

### PR DESCRIPTION
**Pull request by Panquesito7**
Compatibility with MT 5.0.0 and above:

- Replace deprecated functions with newer ones.
- Use `mod.conf` for dependencies and description.
  - `depends.txt` and `description.txt` are kept for compatibility with MT 0.4.16 and above.

### CODE REVIEW

File: init.lua [hunk 1]
```diff
@@ -1,13 +1,16 @@
--- Minetest 0.4.12+ mod: chat2
--- chat2 is mod for minetest game created by Andrey. It's purpose is to improve or replace game built in chat.
---  + Higlight some messages in different colors - nearby talk, PM messages, messages with your name in it, shouts(!)
---  + Also can show all regular chat messages (use "/chat2 *" command)
---  + Switch on/off (use "/chat2" command)
---  + Old messages dissapear after some time
---  - Problems with unicode symbols on some clients
+--[[ 
+Minetest 5.0.0+ mod: chat2
```
Comment (MoNTE48) [inline init.lua]: What of your changes break compatibility with MT 0.4?

Comment (Panquesito7) [reply init.lua]: See commit [`50009a2`](https://github.com/Panquesito7/chat2/commit/50009a2e455e029ddfc513c63719258da4617766)

Comment (MoNTE48) [reply init.lua]: Well, if saving a few bytes due to debug.txt is the benefit of broken compatibility... You're a real troll 😬

Comment (Panquesito7) [reply init.lua]: It is now compatible with MT 0.4.16 and above. 🙂

**Pull request merged**